packages/core/src/lib/components/image/image-viewer.tsx

```diff
@@ -106,6 +106,26 @@ const ImageViewer: React.FC<ImageViewerProps> = ({
     [urls, setCurrentImageIndex, setIsOpened],
   );
 
+  useEffect(() => {
+    const styleElement = document.createElement("style");
+
+    if (isOpened) {
+      document.body.setAttribute("data-scroll-locked", "true");
+      const gap = getGapWidth();
```

Comment:
p2: you should capsulize this hook.

I suggest use-prevent-scroll

Reply:
I checked this review.